core/src/std_packages.rs

```diff
@@ -174,41 +260,125 @@ pub fn native_block_content<T, U>(
-        Ok(Element::Module {
-            name: "__text".to_string(),
-            args: Default::default(),
-            body: list.join(","),
-            inline: false,
-            id: id.clone(),
-        })
-    } else {
-        Ok(Element::Compound(vec![]))
+    let name = args.get("name").unwrap().as_str().unwrap();
+
+    match ctx.state.variables.get(name) {
+        Some(value @ variables::Value::Constant(_)) => {
+            Ok(text_element(value.to_string(), id.clone()))
+        }
+        Some(value) => Err(CoreError::TypeMismatch {
+            name: name.to_string(),
+            expected_type: VarType::Constant,
+            present_type: value.get_type(),
+        }),
+        None => Ok(Element::Compound(vec![])), // TODO: give a error or warning
+    }
+}
+
+/// Push a value to a list
+pub fn list_push<T, U>(
+    ctx: &mut Context<T, U>,
+    value: &str,
+    args: HashMap<String, ArgValue>,
+    _: bool,
+    _: &OutputFormat,
+    _: &GranularId,
+) -> Result<Element, CoreError> {
+    let name = args.get("name").unwrap().as_str().unwrap();
+    ctx.state.variables.list_push(name, value)?;
+    Ok(Element::Compound(vec![]))
+}
+
+/// Read a list of values
+pub fn list_read<T, U>(
+    ctx: &mut Context<T, U>,
+    _: &str,
+    args: HashMap<String, ArgValue>,
+    _: bool,
+    _: &OutputFormat,
+    id: &GranularId,
+) -> Result<Element, CoreError> {
+    let name = args.get("name").unwrap().as_str().unwrap();
+
+    match ctx.state.variables.get(name) {
+        Some(value @ variables::Value::List(_)) => Ok(text_element(value.to_string(), id.clone())),
+        Some(value) => Err(CoreError::TypeMismatch {
+            name: name.to_string(),
+            expected_type: VarType::List,
+            present_type: value.get_type(),
+        }),
+        None => Ok(Element::Compound(vec![])), // TODO: give a error or warning
+    }
+}
+
+/// Add a string to a set
+pub fn set_add<T, U>(
+    ctx: &mut Context<T, U>,
+    value: &str,
+    args: HashMap<String, ArgValue>,
+    _: bool,
+    _: &OutputFormat,
+    _: &GranularId,
+) -> Result<Element, CoreError> {
+    let name = args.get("name").unwrap().as_str().unwrap();
+    ctx.state.variables.set_add(name, value)?;
+    Ok(Element::Compound(vec![]))
+}
+
+/// Read a list of values
+pub fn set_read<T, U>(
+    ctx: &mut Context<T, U>,
+    _: &str,
+    args: HashMap<String, ArgValue>,
+    _: bool,
+    _: &OutputFormat,
+    id: &GranularId,
+) -> Result<Element, CoreError> {
+    let name = args.get("name").unwrap().as_str().unwrap();
+
+    match ctx.state.variables.get(name) {
+        Some(value @ variables::Value::Set(_)) => Ok(text_element(value.to_string(), id.clone())),
+        Some(value) => Err(CoreError::TypeMismatch {
+            name: name.to_string(),
+            expected_type: VarType::Set,
+            present_type: value.get_type(),
+        }),
+        None => Ok(Element::Compound(vec![])), // TODO: give a error or warning
```

Comment:
Something to do?

Reply:
Out of our previous discussions, I have really liked the idea of defaulting to an empty list/set when reading something that hasn't been declared, while you have been very keen on keeping it undefined. I think the most common thing people would expect when doing `[set-read] imports` is not to give a big error but rather not print any authors if there aren't any. I know you disagree, but it might be a good idea to add an argument to default to an empty set if absent?